### ISSUE TITLE
Consider moving `epi_shift()` from `epipredict` here, exporting, maybe generalizing

### ISSUE THREAD

**Issue by brookslogan** [cmu-delphi/epiprocess#457]
This may make it easier to work with lags (correctly) with fable or other non-epipredict forecasters.  I'm not sure if fable AR works properly if you just have e.g. lag(exog, 7L) in the formula if there are gaps.

@dajmcdon does this sound like a good idea?

**Comment by brookslogan** [comment]
I realize I didn't test the right thing above.  The original issue I was thinking of was raised by Richard, where you want to use lags of an exogeneous variable as predictors.

**Comment by dajmcdon** [comment]
Maybe I'm confused by "handle". That warning above is fatal: it doesn't fit anything and returns a NULL model object. This is the case for both explicit and implicit gaps. So implicit, it tells you to fill it, and explicit, it fails at fit time due to the NA. Is that OK for this purpose?

**Comment by brookslogan** [comment]
Yep, that all sounds fine. "Handle" = generate error is OK.  So the use case for `epi_shift()` isn't fable after all.  It'd probably be when we want to use some non-fable non-epipredict thing and want to "properly" take lags accounting for the `time_value`.  (Of course, this thing needs to be able to accept NAs in the training data and accept/avoid them in the test data.  I'd imagine non-epipredict-based but still regression-based approaches would be one example.)
